server/src/controllers/admin.ts

```diff
@@ -106,6 +111,38 @@ export const validateSession = async (req: Request, res: Response) => {
   }
 }
 
+export const authenticateRequest = async (req: Request, res: Response, next: NextFunction) => {
```

Comment:
this function sucks...i tried using convention for authenticated rest requests, (an auth token) .. since we use emails and tokens manage sessions, we still need the current user email. We can just throw that in the body on a get request and it should be fine. Sucks bc it breaks convention but we would have to change how we manage sessions and we probably dont have time for that rn...